File: typescript/stack.ts
export class Stack {
    private data: any[] = [];

    public top(): any {
        return this.data.at(-1);
    }

    public isEmpty(): boolean {
        return this.data.length === 0;
    }

    public push(elem: any) {
        this.data.push(elem);
    }

    public pop() {
        if(this.isEmpty()) {
            return undefined;
        }
        return this.data.pop();
    }

    public printStack() {
        console.log(this.data)
    }
}




const stack = new Stack()

stack.push(1)
stack.push(2)
stack.push(3)

stack.printStack()
console.log(stack.pop())

console.log(stack.pop())

console.log(stack.pop())

console.log(stack.pop())

File: python/stack.py
class Stack:
    data = []

    def top(self):
        if self.empty():
            return None
        return self.data[-1]

    def push(self, x):
        self.data.append(x)

    def empty(self):
        return len(self.data) == 0

    def pop(self):
        if self.empty():
            return None
        return self.data.pop()
    
    def printStack(self):
            print(self.data)


stack = Stack()

stack.push(1)
stack.push(2)
stack.push(3)

stack.printStack()
print(stack.pop())

print(stack.pop())

print(stack.pop())

print(stack.pop())
